### DISCUSSION TITLE
Script Host Detection

### DISCUSSION THREAD

**Discussion by rkwapisz** [YimMenu/YimMenu#3057] [General]
One problem I've been running into (which I cannot verify since I can't detect script hosts) is that certain players appear to be joining sessions as script host, and then bricking the entire server. I don't know if this is intentionally malicious behavior with a menu, or if the script host just happens to be super high latency or running on a potato PC, given the P2P system.

Some common patterns that I've observed include:
- Inability to call personal vehicles, resulting in the mechanic calling you with his error message
- Inability to start/complete public events such as Import/Export, drug vehicles, etc.
- Inability to enter/exit vehicles or interiors
- Inability to interact with just about anything (safes, passed out guards, etc.)

In a recent case, I had a person join my session which immediately started causing severe issues. I guessed correctly that it was the most recent join, and host kicked them. Everything then went back to normal.

Is there a practical way to detect a script host in a session to more easily identify poor script hosts?

**Comment by xesdoog** [comment]
I second this. I've noticed it happening multiple times and it also breaks one of my scripts (some spawned peds would freeze and the script loses control over them).

**Comment by lonelybud** [comment]
its there in debug section under scripts

**Reply by rkwapisz** [reply]
This?

![image](https://github.com/YimMenu/YimMenu/assets/13259220/96b100b3-2b8f-48f8-8601-e0eaafd36e21)

Where is this going to tell me who the freemode script host is?

**Reply by lonelybud** [reply]
threads my bad

**Comment by rkwapisz** [comment]
Thank you @lonelybud.

![image](https://github.com/YimMenu/YimMenu/assets/13259220/173aa3d0-cf4b-40e5-9b46-2bebbe53b16c)
